docs/csharp/roslyn-sdk/syntax-visualizer.md

```diff
@@ -0,0 +1,138 @@
+
+Create a new project using the **File** –> **New Project** command. You can create either a VB or C# project. When Visual Studio opens the main code file for this project, the visualizer displays the syntax tree for it. You can open any existing C# / VB file in this Visual Studio instance, and the visualizer displays that file's syntax tree. If you have multiple code files open inside Visual Studio, the visualizer will always display the syntax tree for the currently active code file, the code file that has keyboard focus.
+
+![Visualizing a C# syntax tree](media/fig3.png)
+
+![Visualizing a VB syntax tree](media/fig4.png)
+
+As shown in the preceding images, the visualizer tool window displays the syntax tree at the top and a property grid at the bottom. The property grid displays the properties of the item that is currently selected in the tree, including the .NET *Type* and the *Kind* (SyntaxKind) of the item.
+
+Syntax trees comprise three types of items – *nodes*, *tokens*, and *trivia*. You can read more about these types in the [.NET Compiler Platform SDK Overview document](compiler-api-model.md). Items of each type are represented using a different color. Click on the button titled ‘Legend’ for an overview of the colors used.
+
+![Legend](media/fig5.png)
+
+Each item in the tree also displays its own **span**. The **span** is the indices of that node in the text file.  In the preceding C# example, the selected “UsingKeyword [0..5)” token has a **Span** that is 5 characters wide, [0..5). The "[..)" notation means that the starting index is part of the span, but the ending index is not.
+
+There are two ways two navigate the tree:
+* Expand or click on items in the tree. The visualizer automatically selects the text corresponding to this item’s Span in the code editor.
+* Clicking or Select text in the code editor. In the preceding VB example, if you select the line containing "Module Module1" in the code editor, the visualizer automatically navigates to the corresponding ModuleStatement node in the tree. 
+
+The visualizer highlights the item in the tree whose span best matches the span of the text selected in the editor.
+
+The visualizer refreshes the tree to match modifications in the active code file. Add a call to `Console.WriteLine()` inside `Main()`. As you type, the visualizer refreshes the tree.
+
+Pause typing once you have typed `Console.`. The tree has some items colored in pink. At this point, there are errors (also referred to as ‘Diagnostics’) in the typed code. These errors are attached to nodes, tokens, and trivia in the syntax tree. The visualizer shows you which items have errors attached to them highlighting the background in pink. You can inspect the errors on any item colored pink by hovering over the item. The visualizer will only display syntactic errors - those errors related to the syntax of the typed code – it will not display any semantic errors.
```

Comment:
*the visualizer only displays syntactic errors [...] it does not display any semantic errors.